project/src/demo/puzzle/critter/carrot_demo.gd

```diff
@@ -0,0 +1,17 @@
+extends Node
+## A demo which shows off the carrot puzzle critter.
```

Comment:
Carrot is a 'puzzle vegetable'...? or a critter? Be consistent.

Reply:
Changed to 'critter' for consistency. Intuitively there may eventually be different groups for 'animals', 'demons', 'vegetables', and 'candies' but for now they are all just critters. There are only two!